src/components/tiles/AddressTile2.tsx

```diff
@@ -218,11 +242,18 @@ export class AddressTileComponent extends React.PureComponent<Props, State> {
   }
 
   render() {
-    const { fioToAddress, recipientAddress, lockInputs, theme, title } = this.props
+    const { fioToAddress, recipientAddress, lockInputs, theme, title, currencyWallets, coreWallet } = this.props
+    const { pluginId, currencyCode } = coreWallet.currencyInfo
     const { loading } = this.state
     const styles = getStyles(theme)
     const copyMessage = this.state.clipboard ? `${s.strings.string_paste}: ${this.state.clipboard}` : null
     const tileType = loading ? 'loading' : !!recipientAddress && !lockInputs ? 'delete' : 'static'
+    const tokenId: string | undefined = getTokenId(this.props.account, pluginId, currencyCode)
+    const canSelfTransfer: boolean = Object.keys(currencyWallets).some(
+      walletId =>
+        (walletId !== coreWallet.id && currencyWallets[walletId].type === coreWallet.type) ||
```

Comment:
After staring carefully at this logic, I think it works, but it's not as readable as I would like. I also think there is a danger that two chains might end up with the same tokenId for different assets (I'm thinking of EOS clones, specifically), and we might show the button when it doesn't make sense.

So, I would propose:

```ts
walletId => {
  if (walletId === coreWallet.id) return false
  if (currencyWallets[walletId].type !== coreWallet.type) return false
  if (tokenId == null) return true
  return currencyWallets[walletId].enabledTokenIds.includes(tokenId)
}
```

That way we ensure we are on the same chain first, before we do anything about tokens.

If you still want a one-liner:


```ts
walletId => 
  walletId !== coreWallet.id &&
  currencyWallets[walletId].type === coreWallet.type && (
    tokenId == null ||
    currencyWallets[walletId].enabledTokenIds.includes(tokenId)
  )
```